### PR TITLE
Support spec v1beta1 resources

### PR DESCRIPTION
**Pull request by sadlerap**
Implements support for spec/v1beta1 resources.

# Changes

/kind enhancement

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#docs) 
  included if any changes are user facing
- [ ] [Tests](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#tests)
  included if any functionality added or changed. For bugfixes please include tests that can catch regressions
- [ ] All acceptance test scenarios included in the PR which verifies a bugfix or a requested feature reported by a non-member are tagged with `@external-feedback` tag.
- [ ] Follows the [commit message standard](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#commits)

### CODE REVIEW

**Review by baijum — CHANGES_REQUESTED** [review]

File: PROJECT [hunk 1]
```diff
@@ -50,4 +50,26 @@ resources:
   kind: BindableKinds
   path: github.com/redhat-developer/service-binding-operator/apis/binding/v1alpha1
   version: v1alpha1
+- api:
+    crdVersion: v1
+    namespaced: true
+  domain: operators.coreos.com
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply PROJECT:56]: - I _think_ this was autogenerated by `kubebuilder`, not entirely certain.
- The `v1alpha3` entries also use `domain: operators.coreos.com`, I'm not entirely certain why.

File: apis/spec/v1beta1/servicebinding_types.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Conditions are the conditions of this ServiceBinding
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// Binding exposes the projected secret for this ServiceBinding
+	Binding *ServiceBindingSecretReference `json:"binding,omitempty"`
+}
+
+// +kubebuilder:object:root=true
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply apis/spec/v1beta1/servicebinding_types.go:105]: According to [this](https://book.kubebuilder.io/reference/markers/crd.html), the default is indeed `Namespaced`.

Fixed in 0ceefae0.

File: main.go [hunk 3]
```diff
@@ -47,6 +47,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	specv1alpha3 "github.com/redhat-developer/service-binding-operator/apis/spec/v1alpha3"
```
[the remark this answers is not in the document]
Comment (baijum) [reply main.go:49]: We can remove these files:
- apis/spec/v1alpha3/clusterworkloadresourcemapping_webhook.go
- apis/spec/v1alpha3/clusterworkloadresourcemapping_webhook_test.go
- apis/spec/v1alpha3/servicebinding_webhook.go
- apis/spec/v1alpha3/servicebinding_webhook_test.go
- apis/spec/v1alpha3/suite_test.go

[the remark this answers is not in the document]
Comment (sadlerap) [reply main.go:49]: Fixed with 5c3f3e4.

**Pull request merged**